### PR TITLE
Use Mapbox GL for visualisations in the worked examples

### LINKED ISSUE #526: Re-enable docs notebooks execution during CI
We had to disable execution on CI of the docs notebooks, because they weren't running in a timely enough fashion. This is obviously undesirable, as we want to be sure they continue to work. We should resolve the not-running-fast enough, and re-enable execution asap.

### PR DESCRIPTION
**Pull request by jch1g10**
Closes #526 #528 #711

### I have:

- [x] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [ ] Added tests for any new additions
- [x] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [ ] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [x] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

This PR replaces the use of Folium with Mapbox GL for producing choropleths and heatmaps in the worked examples.

Running `pipenv run build` now also pre-caches flowmachine queries by executing `docs/cache_queries.py`, meaning that the example notebooks can run quickly enough to be executed during the CI docs build.

Note: Mapbox requires an API token for producing the visualisations. The dockerised worked examples are set up to read this from the `MAPBOX_ACCESS_TOKEN` env var. I haven't made it mandatory to set this env var before spinning up the worked examples container, because some people may want to run parts of the example notebooks without having to sign up for a mapbox account, but I have added a note to the docs explaining that a token is required to produce the maps.

Another note: `mapboxgl-jupyter` is not yet as full-featured as folium, and doesn't currently include some features we were previously using, such as multi-layered choropleths and time sliders. There are open issues for both of these features and more, though (see https://github.com/mapbox/mapboxgl-jupyter/issues/34, https://github.com/mapbox/mapboxgl-jupyter/issues/82), so we should continue to monitor this.

### CODE REVIEW

File: docs/cache_queries.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This file is executed in build.sh before building the docs.
+# The purpose is to speed up execution of the example notebooks,
+# by caching some of the queries beforehand.
```
Comment (maxalbert) [inline docs/cache_queries.py]:
```suggestion
# by caching some of the queries beforehand (which is currently
# not possible using flowclient directly).
```

**Comment by jch1g10** [comment]
Running
`CI=true CIRCLE_SHA1=a010142b007b32d5f63012a16e295536a7645fbc bash <(curl -s https://raw.githubusercontent.com/Flowminder/FlowKit/a010142b007b32d5f63012a16e295536a7645fbc/quick_start.sh) examples`
should spin up the demo system with the worked examples.

**Comment by maxalbert** [comment]
@jch1g10 This command does indeed spin up the demo system with the worked examples, but running the notebooks fails because it spins up `flowdb_testdata` rather than `flowdb_synthetic_data`, so the flows calculation returns the following error:
```python
FlowclientConnectionError: Something went wrong. API returned with status code 400.
Error message: 'Unable to create query object.'.
Payload: {'exception': 'No data for date: 2016-01-08 - 2016-01-09'}
```
Do you think there is an easy fix for this? If not, I'd rather address it in a separate PR.

**Comment by jch1g10** [comment]
> @jch1g10 This command does indeed spin up the demo system with the worked examples, but running the notebooks fails because it spins up `flowdb_testdata` rather than `flowdb_synthetic_data`, so the flows calculation returns the following error:
> 
> ```python
> FlowclientConnectionError: Something went wrong. API returned with status code 400.
> Error message: 'Unable to create query object.'.
> Payload: {'exception': 'No data for date: 2016-01-08 - 2016-01-09'}
> ```
> 
> Do you think there is an easy fix for this? If not, I'd rather address it in a separate PR.

Oops! Should use the synthetic data now, with the latest commit:
```
CI=true CIRCLE_SHA1=7c868eb4c71994241fa1ce1e9b3a98d1186b604a bash <(curl -s https://raw.githubusercontent.com/Flowminder/FlowKit/7c868eb4c71994241fa1ce1e9b3a98d1186b604a/quick_start.sh) examples
```
(once CI finishes)

**Comment by maxalbert** [comment]
Thanks! I tried to run the worked examples using this command. I can execute part of the Flows notebook, but sadly I get an internal error. I don't think this has anything to do with this PR, though, so I'm happy to merge this.

**Pull request merged**